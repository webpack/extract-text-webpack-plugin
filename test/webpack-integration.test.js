/* eslint-disable
  global-require,
  import/no-dynamic-require,
  consistent-return
*/
import fs from 'fs';
import path from 'path';
import webpack from 'webpack';
import glob from 'glob';
import ExtractTextPlugin from '../src';

const cases = process.env.CASES
  ? process.env.CASES.split(',')
  : fs.readdirSync(path.join(__dirname, 'cases'));

describe('Webpack Integration Tests', () => {
  cases.forEach((testCase) => {
    it(testCase, (done) => {
      let options = { entry: { test: './index.js' } };
      const testDirectory = path.join(__dirname, 'cases', testCase);
      const outputDirectory = path.join(__dirname, 'js', testCase);
      const configFile = path.join(testDirectory, 'webpack.config.js');

      if (fs.existsSync(configFile)) {
        options = require(configFile);
      }
      options.context = testDirectory;
      if (!options.module) options.module = {};
      if (!options.module.rules) {
        options.module.rules = [
          { test: /\.txt$/, loader: ExtractTextPlugin.extract('raw-loader') },
        ];
      }
      if (!options.output) options.output = { filename: '[name].js' };
      if (!options.output.path) options.output.path = outputDirectory;
      if (process.env.CASES) {
        // eslint-disable-next-line
        console.log(
          `\nwebpack.${testCase}.config.js ${JSON.stringify(options, null, 2)}`
        );
      }

      webpack(options, (err, stats) => {
        if (err) return done(err);
        if (stats.hasErrors()) return done(new Error(stats.toString()));
        const testFile = path.join(outputDirectory, 'test.js');
        if (fs.existsSync(testFile)) {
          require(testFile)(suite);
        }
        const expectedDirectory = path.join(testDirectory, 'expected');

        glob
          .sync('**/*.@(txt|css)', { cwd: outputDirectory })
          .forEach((file) => {
            const filePath = path.join(expectedDirectory, file);
            const actualPath = path.join(outputDirectory, file);
            expect(fs.existsSync(filePath)).toBeTruthy();
            expect(fs.readFileSync(actualPath, 'utf-8')).toEqual(
              fs.readFileSync(filePath, 'utf-8')
            );
            expect(fs.readFileSync(actualPath, 'utf-8')).toMatchSnapshot();
          });
        done();
      });
    });
  });
});
