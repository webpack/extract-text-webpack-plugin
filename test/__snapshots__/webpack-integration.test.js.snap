// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack Integration Tests chunk-modules-css-wrong-order 1`] = `
".block {
  color: tomato;
}
.App {
  color: black;
}
"
`;

exports[`Webpack Integration Tests chunk-modules-nested-ordered-by-id 1`] = `
"body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}
.block {
	color: tomato;
}
.App {
	color: black;
}
"
`;

exports[`Webpack Integration Tests chunk-modules-ordered-by-id 1`] = `
".block {
	color: tomato;
}
.App {
	color: black;
}
"
`;

exports[`Webpack Integration Tests common-async 1`] = `
"a
b
c
"
`;

exports[`Webpack Integration Tests empty-chunk 1`] = `
"a
"
`;

exports[`Webpack Integration Tests merging-chunk 1`] = `
"a
b
c
"
`;

exports[`Webpack Integration Tests merging-multiple-entries 1`] = `
"a
b
"
`;

exports[`Webpack Integration Tests merging-multiple-entries 2`] = `
"a
c
"
`;

exports[`Webpack Integration Tests merging-multiple-entries-shared 1`] = `
"a
"
`;

exports[`Webpack Integration Tests merging-multiple-entries-shared 2`] = `
"a
"
`;

exports[`Webpack Integration Tests multiple-entries 1`] = `
"a
b
"
`;

exports[`Webpack Integration Tests multiple-entries 2`] = `
"a
c
"
`;

exports[`Webpack Integration Tests multiple-entries-filename 1`] = `
"a
b
"
`;

exports[`Webpack Integration Tests multiple-entries-filename 2`] = `
"a
c
"
`;

exports[`Webpack Integration Tests nested 1`] = `
"a
b
c
d
"
`;

exports[`Webpack Integration Tests optimize-tree 1`] = `
"index
a
b
"
`;

exports[`Webpack Integration Tests order-undefined-error 1`] = `
"._3QiTkN7JPds2C9Bq73MpOX {
}
._3dtwYD12yqGZUJ-uPQLu9z {
}
._32HVXkGocfWZMaHoXEFkED {
}
.hbZI7DlQBB9VYNCgL20kL {
}
"
`;

exports[`Webpack Integration Tests simple 1`] = `
"a
b
"
`;

exports[`Webpack Integration Tests simple-query-object 1`] = `
"body {
	correct: a;
}
body {
	correct: b;
}

/*# sourceMappingURL=file.css.map*/"
`;

exports[`Webpack Integration Tests simple-queryless-object 1`] = `
"body {
	correct: a;
}
body {
	correct: b;
}

/*# sourceMappingURL=file.css.map*/"
`;

exports[`Webpack Integration Tests splitted-chunk 1`] = `
"a
"
`;

exports[`Webpack Integration Tests splitted-chunk-import 1`] = `
"a
"
`;

exports[`Webpack Integration Tests splitted-chunk-import-allchunks 1`] = `
"a
b
"
`;

exports[`Webpack Integration Tests splitted-multiple-entries 1`] = `""`;

exports[`Webpack Integration Tests splitted-multiple-entries 2`] = `""`;

exports[`Webpack Integration Tests string-export-with-optimize-module-order 1`] = `
"a
b
"
`;
